pkg/apis/discovery/validation/validation.go

```diff
@@ -221,7 +222,7 @@ func validateHints(endpointHints *discovery.EndpointHints, fldPath *field.Path)
 		return allErrs
 	}
 
-	zoneNames := sets.String{}
+	zoneNames := sets.Set[string]{}
```

Comment:
Given how often EndpointSlices are updated, we should try to minimize wasteful allocations.  A set is a map which is several allocations under the covers.  Could I talk you into doing this as `make([]string, 0, len(endpointHints.ForZones))` and doing a linear search?  (with comment on why).  Or even a `[maxZoneHints]string`  and `nzones int`?

Reply:
Done (with the slice version so I could use `slices.Contains`), and updated the port name validation as well. I left `supportedAddressTypes` and `supportedPortProtocols` using sets since those sets are just created once.